oneoffs/cbt_models.py

```diff
@@ -0,0 +1,169 @@
+#!/usr/bin/env python3
+
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Write Minigo model records to Bigtable.
+
+This is used to insert models data into cbt model table from gcs training dir.
+"""
+
+import sys
+sys.path.insert(0, '.')
+
+import re
+
+from absl import app, flags
+from google.cloud import bigtable
+from google.cloud.bigtable import row_filters
+from tensorflow import gfile
+from tqdm import tqdm
+
+from bigtable_input import METADATA, TABLE_STATE
+
+
+flags.DEFINE_string(
+    'model_glob', None,
+    'Glob for .meta model files to backfill into models bigtable.')
+
+flags.mark_flags_as_required([
+    'model_glob', 'cbt_project', 'cbt_instance', 'cbt_table'
+])
+
+FLAGS = flags.FLAGS
+
+# Constants
+
+# TODO(sethtroisi): Move to cluster or somewhere.
+MODEL_ROW = 'm_{}_{:0>10}'
+MODEL_NAME = b'model'
+
+MODEL_RUN_NAME_RE = re.compile(
+    r'^gs://.*(v[1-9][0-9]?)-(19|9)/models/(([0-9]{6})-([a-z-]*)).meta$')
+
+#### START ####
+
+
+def read_existing_models(bt_table):
+    """Return model names from each existing record."""
+    # TODO(dtj): Is there a clean way to read just row_keys.
+    rows = bt_table.read_rows(filter_=row_filters.ColumnRangeFilter(
+        METADATA, MODEL_NAME, MODEL_NAME))
+    return [row.row_key.decode() for row in rows]
+
+
+def canonical_name(model_path):
```

Comment:
This function's name feels at odds with what it does. It's not creating a canonical name, but decomposing one. Perhaps `parse_canonical_model_path`, or `parse_model_components`?

Reply:
totally agree, changing to parse_model_components.